### DISCUSSION TITLE
How to invoke a business rule in a root object or from a list object when child is deleted from a child list

### DISCUSSION THREAD

**Discussion by CProsRodneyWorkman** [MarimerLLC/csla#2468] [Questions]
I have client side business actions that need to performed to other children in child list object when one of the children are deleted.  The only way I can find to solve this is by detecting the delete from the UI and then have the UI invoke a post delete method.  

Just for grins I tried to invoke a business rule from a rule registered in the root object using the child list object property as the trigger for the rule but that did not fire.  

BusinessRules.AddRule(new PriceEscalatorProductEditListChangedRule(**PriceEscalatorProductEditListProperty**));

Any suggestions?

**Comment by rockfordlhotka** [comment]
The correct answer is typically to handle the child changed event in the parent of the collection, and in that event handler you can use the eventargs to determine what changed and whether to run rule(s).

Normally you'll also have some method(s) on the child collection and/or child business types so the parent can ask them to run their rules. Or if the rule is at the level of the parent, just call `BusinessRules.CheckRules()` in the parent.

**Reply by hurcane** [reply]
I would add that since the rule is defined in the parent class, you can run BusinessRules.CheckRules(PriceEscalatorProductEditListProperty) to only run the rules associated with the list property, and not run all rules in the parent business object.
